go/pkg/array/chunk.go

```diff
@@ -0,0 +1,16 @@
+package array
```

Comment:
Is there any reason why we don't use std https://pkg.go.dev/slices#Chunk? 

This also gives us iterator so no need for additional range over our result type

```go
for chunk := range slices.Chunk([]int{1,2,3,4,5}, 2) {
    fmt.Println(chunk) // [1 2], [3 4], [5]
}
```